app/frontend/components/pages/login/loginPageSidebar.tsx

```diff
@@ -164,23 +166,16 @@ const FileContent = () => (
   </div>
 )
 
-interface Props {
-  authMethod: '' | 'mnemonic' | 'hw-walet' | 'file'
+const LoginPageSidebar = () => {
+  const {authMethod} = useSelector((state: State) => ({authMethod: state.authMethod}))
+  return (
+    <aside className="sidebar">
+      {authMethod === AuthMethodEnum.INITIAL && <InitialContent />}
```

Comment:
This seems like a good opportunity to leverage case exhaustiveness checking by typescript: https://stackoverflow.com/questions/59685270/typescript-ensure-all-cases-are-handled-for-union-type

Reply:
This solution is seems a little bit hard to pull off with out current settings. It implicitly assumes strictNullChecks tsconfig option to be set "true". If you go ahead to the playground and turn the option to off, this solution no longer works.

If I turn this on, it introduces a billion errors from all parts of code. The other solution - with strictNullChecks off - is to have an assert at the end of such exhaustive functions https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript . But imho that pollutes the codebase if we were to do that in every such an exhaustive checking function. What do you think?

Reply:
I introduced the assert function and exhaustive switches for the time being. I handled the most obvious auth method functionalities, although it could be refactored further for DropdownItems and AuthTabs, but that would require iterating a typescript enum, which is not that pretty.